controller/behavior/denormalizeable.php

```diff
@@ -0,0 +1,188 @@
+<?php
+/**
+ * @package     LOGman
+ * @copyright   Copyright (C) 2011 - 2016 Timble CVBA. (http://www.timble.net)
+ * @license     GNU GPLv3 <http://www.gnu.org/licenses/gpl.html>
+ * @link        http://www.joomlatools.com
+ */
+
+/**
+ * Denormalizeable Controller Behavior
+ *
+ * Allows for de-normalization of activities entries
+ *
+ * @author  Arunas Mazeika <https://github.com/amazeika>
+ * @package Joomlatools\Component\LOGman
+ */
+class ComActivitiesControllerBehaviorDenormalizeable extends KControllerBehaviorAbstract
+{
+    /**
+     * The resources controller
+     *
+     * @var KModelInterface
+     */
+    protected $_controller;
+
+    public function __construct(KObjectConfig $config)
+    {
+        parent::__construct($config);
+
+        $this->_controller = $config->controller;
+    }
+
+    protected function _initialize(KObjectConfig $config)
+    {
+        $config->append(array(
+            'controller' => 'resource'
+        ));
+
+        if ($this->getIdentifier()->getPackage() != 'activities')
+        {
+            $aliases = array(
+                'com:activities.controller.resource' => array(
+                    'path' => array('controller'),
+                    'name' => 'resource'
+                )
+            );
+
+            foreach ($aliases as $identifier => $alias)
+            {
+                $alias = array_merge($this->getMixer()->getIdentifier()->toArray(), $alias);
+
+                $manager = $this->getObject('manager');
+
+                // Register the alias if a class for it cannot be found.
+                if (!$manager->getClass($alias, false)) {
+                    $manager->registerAlias($identifier, $alias);
+                }
+            }
+        }
+
+        parent::_initialize($config);
+    }
+
+    protected function _afterAdd(KControllerContextInterface $context)
+    {
+        $entity = $context->result;
+
+        if ($entity->action != 'delete')
+        {
+            $context = $this->getMixer()->getContext();
+
+            $context->entity = $entity;
+
+            $this->denormalize($context);
```

Comment:
We can keep it BREAD only, no problem. The behavior will still work the same by removing the denormalize action and moving it all to the after.add command handler.

> I would also consider the flow and who is responsible for the denormalisation of the data, you can make this the responsability of this behavior, you can also move this responsability to the resource controller itself, if the resource controller receive an ComActivitiesActivityInterface he would denormalize it.

I really rather keeping the denormalization process in the behavior. The resource controller should just receive the data and create the resource. Then expose it.

> It feels like you are trying to do a bit too much in this behaviour, making it a bit too flexible from the start. Is there are reason why the resource controller needs to be configurable? If not, I would just hard code it.

This controller is configurable and by default points to the same package of the subject (controller implementing it). This way you can have the DB layer sitting on the component implementing the behavior. This can be overridden by providing a full identifier so that it uses the resource controller on com:activities. Hardcoding it would mean pointing it to com:activities directly no?, this wouldn't work if I want the resources table to sit in the component.

> Finally, you consider if adding the activity to the resource table is something that should happen after it has been added to the activities, or if it should happen during logging.

After the activity gets logged, then we should denormalize.

If you agree, I'll just get rid of the denormalize action and handle the denormalization process in the after.add handler.

Thoughts?